sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -1458,17 +1458,18 @@ static void _handle_rtr_timeout(_nib_dr_entry_t *router)
 {
     if ((router->next_hop != NULL) && (router->next_hop->mode & _DRL)) {
         _nib_offl_entry_t *route = NULL;
-        unsigned iface = _nib_onl_get_if(router->next_hop);
-        ipv6_addr_t addr = router->next_hop->ipv6;
+        _nib_onl_entry_t *next_hop = router->next_hop;
 
         _nib_drl_remove(router);
-        /* also remove all routes to that router */
+        /** The Router Lifetime applies only to
+            the router's usefulness as a default router; it
+            does not apply to information contained in other
+            message fields or options. Options that need time
+            limits for their information include their own
+            lifetime fields.
+            @see [RFC4861]https://datatracker.ietf.org/doc/html/rfc4861#section-4.2 */
```

Comment:
How does this comment relate to the fix below? Or should it even? Below you just seem to apply pointer arithmetics, which is a nice optimization, but the comparison has nothing to do with RA fields...

Reply:
The comment expresses the opposite of the previous comment.
Offlink entries are _PFX, _DC, _FT.
Before, all associated prefixes were deleted. But prefixes have their own lifetime.
The _DC flag is cleared in `_nib_drl_remove`.
Assume any flag would be set in `route->mode`. That means it is present in some data structure.
If the mode is just set to _EMPTY and if `_nib_offl_clear()` simply memsets() the entry, it is not properly deleted.
An offlink entry must be deleted properly with (`_nib_pl_remove`, `_nib_ft_remove`, `_nib_dc_remove`).
I remember when I was debugging at that time I saw that the memset was brutally overwriting present entries.